app/services/zendesk_statistics_service.rb

```diff
@@ -0,0 +1,85 @@
+require 'csv'
+
+class ZendeskStatisticsService
+  class << self
+    delegate :send!, to: :new
+  end
+
+  def initialize; end
+
+  def csv_generator
+    CSV.generate(headers: true) do |csv|
+      csv << ['Category',
+              'Title',
+              'Description',
+              'Content',
+              'Usage 1hr',
+              'Usage 24hr',
+              'Usage 7d',
+              'Usage 30d',
+              'Created',
+              'Last updated',
+              'Set tags',
+              'Add tags',
+              'Remove tags']
+
+      all_macros.all! do |macro|
+        if macro.active
+          csv << [
+            format_category(macro.title),
+            format_title(macro.title),
+            macro.description,
+            macro.actions.select { |action| action.field == 'comment_value_html' }.first.value,
+            macro.usage_1h,
+            macro.usage_24h,
+            macro.usage_7d,
+            macro.usage_30d,
+            macro.created_at.strftime('%d/%m/%Y %H:%M'),
+            macro.updated_at.strftime('%d/%m/%Y %H:%M'),
+            tags(macro.actions, 'set_tags'),
+            tags(macro.actions, 'current_tags'),
+            tags(macro.actions, 'remove_tags'),
+          ]
+        end
+      end
+    end
+  end
+
+  def filename
+    "zendesk-macros-#{Time.zone.now.strftime('%Y%m%d_%H%M')}.csv"
+  end
+
+  def all_macros
+    @all_macros ||= client.macros.include(:usage_1h).include(:usage_24h).include(:usage_7d).include(:usage_30d)
+  end
+
+private
+
+  def client
+    @client ||= ZendeskAPI::Client.new do |config|
+      config.url = 'https://get-help-with-tech-education.zendesk.com/api/v2'
```

Comment:
Please could you put this in `settings.yml` as well?

Reply:
🤔 I suppose what I could do is create a `ZendeskBase` class and let this service and the other zendesk service inherit from it. What do you think?

Reply:
I think this kind of static URL to an external API belongs in `settings.yml`